README.md

```diff
@@ -203,6 +214,68 @@ Circuit Maintenance Notification #0
 }
 ```
 
+## How to Extend the Library?
+
+Even the library aims to include support for as many providers as possible, it's likely that not all the thousands of NSP are supported and you may need to add support for some new one. Adding a new `Provider` is quite straight forward, and in the following example we are adding support for an imaginary ABCDE provider, that uses HTML notifications.
+
+First step is creating a new file: `circuit_maintenance_parser/parsers/abcde.py`. This file will contain all the custom parsers needed for the provider and it will import the base classes for each parser type from `circuit_maintenance_parser.parser`. In the example, we only need to import `Html` and in the child class implement the methods required by the class, in this case `parse_html()` which will return a `dict` with all the data that this `Parser` can extract. In this case we have to helper methods, `_parse_bs` and `_parse_tables` that implement the logic to navigate the notification data.
+
+```python
+from typing import Dict
+import bs4  # type: ignore
+from bs4.element import ResultSet  # type: ignore
+from circuit_maintenance_parser.parser import Html
+
+class HtmlParserABCDE1(Html):
+    def parse_html(self, soup) -> Dict:
```

Comment:
Probably would be good to document the type of `soup`, as users may not necessarily have prior experience with BS4.